lib/src/impl/realtime/channels.dart

```diff
@@ -59,6 +59,10 @@ class RealtimePlatformChannel extends PlatformObject implements spec.RealtimeCha
     String name,
     dynamic data
   }) async {
+    bool hasAuthCallback = this.ably.options.authCallback!=null;
+    while (hasAuthCallback && this.realtimePlatformObject.authCallbackInProgress) {
```

Comment:
This busy-waiting looks really bad.
I haven't found `hasAuthCallback` being changed anywhere after the while loop, so I assume the while loop is only about `this.realtimePlatformObject.authCallbackInProgress`, therefore I'd write this as
```dart
if(hasAuthCallback) {
  while(this.realtimePlatformObject.authCallbackInProgress) {
    ...
  }
}
```
also instead of busy-waiting 
`realtimePlatformObject` should return a `Future` that completes when no authCallback is in progress anymore.

Reply:
@zoechi Yes, this part needs to be changed completely. I'll try using completers if that fit well here.

Reply:
@zoechi addressed 34417e964349e49ba2071e2ae93316ecf25a8982